File: Practice/src/ProfileRenamer/Scala/GeneralMethods.scala
package ProfileRenamer.Scala

class GeneralMethods {

  val path = FoldersSc.en_US
  val folder = new java.io.File (path);
  val LIST_OF_FILES = folder.listFiles()
  
  
  def fixSpaces (in: String): String = {
    var out = new StringBuilder("");
    for (c <- in) {
      if ( c ==' ' ) {
        out.append("\\ ")
      } else if ( c == '&' ) {
        out.append("\\&")
      } else {
        out.append(c)
      }
    }
    out.toString()
  }
  
  def removeFolder (in: String): String = {
    return in.substring(path.length()+1);
  }
  
  def removeFolder2 (in: String): String = {
    return in.substring(path.length());
  }
}

File: Practice/src/ProfileRenamer/GeneralMethods.java
package ProfileRenamer;

import java.io.File;


public class GeneralMethods {

//	public static String path = "/Users/chekhovs/Desktop/test/";
//	public static String path = Folders.US_profiles.en_US;
	public static final String PATH = Folders.profiles.en_US;
	
	public static final File FOLDER = new File (PATH);
	public static final File[] LIST_OF_FILES = FOLDER.listFiles();
	
	
	public static String fixSpaces (String in) {
		StringBuilder out = new StringBuilder();
		for (int i = 0; i < in.length(); i++) {
			if ( in.charAt(i) == ' ' ) {
				out.append("\\ ");
			} else if ( in.charAt(i) == '&' ) {
				out.append("\\&");
			} else {
				out.append(in.charAt(i));
			}
		}
		return out.toString();
	}
	
	public static String removeFolder (String in) {
		return in.substring(PATH.length()+1);
	}
	
	public static String removeFolder2 (String in) {
		return in.substring(PATH.length());
	}
}